Lib/ufo2ft/featureWriters/kernFeatureWriter2.py

```diff
@@ -251,7 +251,38 @@ def getKerningPairs(font, side1Classes, side2Classes, glyphSet=None, options=Non
                 glyphSet or {},
                 options or SimpleNamespace(**KernFeatureWriter.options),
             )
-            return [KerningPair(pair.side1, pair.side2, pair.value) for pair in pairs]
+            side1toClass: Mapping[tuple[str, ...], ast.GlyphClassDefinition] = {
+                tuple(
+                    glyph
+                    for glyphs in glyph_defs.glyphSet()
```

Comment:
What type is returned from this glyphSet() method? If it is an actual set of strings then the order changes each time, and you’re turning that into a tuple without any sorting

Reply:
From what I can see, it returns tuples.

Reply:
Ah, I needed to sort the incoming kern pairs. The new writer does this as well, as does the old writer below the `if isinstance(font, DesignSpaceDocument):` block.